src/spot_skills/src/spot_skills/planning_scene/load_meshes.py

```diff
@@ -0,0 +1,248 @@
+    if type(mesh) is not Trimesh:
+        raise ImportedMeshTypeError(mesh_path, type(mesh))
+
+    # mesh.show()  # Optionally visualize the mesh
+
+    mesh = mesh.convert_units("meters", guess=True)
+    (min_x, min_y, min_z), (max_x, max_y, _) = mesh.bounds
+
+    x_size = max_x - min_x
+    y_size = max_y - min_y
+
+    translate_by = np.array([-min_x - x_size / 2.0, -min_y - y_size / 2.0, -min_z])
+    normalized_mesh: Trimesh = mesh.apply_translation(translate_by)
+
+    if adjustment_matrix is not None:
+        normalized_mesh = normalized_mesh.apply_transform(adjustment_matrix)
+
+    return normalized_mesh
+
+
+def make_collision_object_msg(object_fields: dict[str, Any]) -> CollisionObjectMsg:
+    """Create a moveit_msgs/CollisionObject message from an object's YAML data.
+
+    Note: This method sets the CollisionObject's `operation` field to ADD.
+
+    :param object_fields: Dictionary from YAML field names to field data
+    :returns: Constructed moveit_msgs/CollisionObject ROS message
+    """
+    collision_object = CollisionObjectMsg()
+
+    # If the relative frame is unspecified, default to "map"
+    collision_object.header.frame_id = object_fields.get("frame", "map")
+
+    pose_xyz_rpy = object_fields.get("pose", [0] * 6)  # Default to identity pose
+    xyz = tuple(pose_xyz_rpy[:3])
+    r_rad, p_rad, y_rad = pose_xyz_rpy[3:]
+
+    # Subframes are defined relative to the object's pose
+    collision_object.pose = create_pose(xyz, r_rad, p_rad, y_rad)
+
+    collision_object.id = object_fields["ID"]
+
+    object_type = object_fields["type"]
+    collision_object.type.key = object_type  # Ignore 'db' field of message
+
+    # Resolve the mesh's package-relative filepath as an absolute path
+    relative_mesh_path = object_fields["mesh_filepath"]
+    mesh_path = resolve_package_path(relative_mesh_path)
+
+    # Check if the mesh should be adjusted during normalization
+    # Example: We may want to axis-align a table mesh within its frame
+    adjustment = object_fields.get("adjust_mesh")
+    if adjustment is None:
+        mesh = load_normalized_mesh(mesh_path)
+    else:
+        adjust_matrix = compose_matrix(
+            translate=adjustment[:3],
+            angles=adjustment[3:],
+        )
+        mesh = load_normalized_mesh(mesh_path, adjust_matrix)
+
+    collision_object.meshes = [trimesh_to_msg(mesh)]
+    collision_object.mesh_poses = [geometry_msgs.msg.Pose()]  # Mesh frame = Object's
+
+    # Create subframes visualizing the bounds of any "placement surface" objects
+    if PutDownSurface.valid_surface_type(object_type):
+        for subframe_name, subframe_pose in get_subframes(mesh).items():
+            collision_object.subframe_names.append(subframe_name)
+            collision_object.subframe_poses.append(subframe_pose)
+
+    collision_object.operation = CollisionObjectMsg.ADD
+
+    return collision_object
+
+
+def trimesh_to_msg(mesh: Trimesh) -> MeshMsg:
```

Comment:
Replace the `Msg` type aliases with the full package's type (for lookup clarity)

Reply:
Done.